test/unit/iso_generator_test.rb

```diff
@@ -0,0 +1,20 @@
+require 'test_plugin_helper'
+
+class ForemanBootdisk::IsoGeneratorTest < ActiveSupport::TestCase
+  include ForemanBootdiskTestHelper
+  setup :setup_bootdisk
+  setup :setup_org_loc
+  setup :setup_subnet
+  setup :setup_host
+
+  setup do
+    @host.build = true
+  end
+
+  test "full host image generation generates via PXELinux type" do
+    @host.expects(:generate_pxe_template).with(:PXELinux)
```

Comment:
Doesn't stubbing this undermine the purpose of the test, to make sure it actually works? If it's stubbed expecting an argument and the method signature changes, the test will never fail.